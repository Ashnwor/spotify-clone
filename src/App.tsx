import { Suspense, lazy, useState } from 'react';
import './styles/App.scss';
import { HashRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Login from './screens/Login';
import UserContext from './context/User';
import Auth from './components/Auth/Auth';
import Loading from './screens/Loading/Loading';

const App: React.FC = () => {
  const [user, setUser] = useState<{ email?: string; name?: string; token?: string }>({});

  const OverviewLazy = lazy(() => import('./screens/Overview'));
  const PlayerLazy = lazy(() => import('./screens/Player'));
  return (
    <UserContext.Provider value={{ user, setUser }}>
      <Router>
        <Switch>
          <Route exact path="/">
            <Redirect to="/login" />
          </Route>
          <Route path="/login">
            <Login />
          </Route>
          <Auth>
            <Route path="/overview">
              <Suspense fallback={<Loading message="Loading Overview" />}>
                <OverviewLazy />
              </Suspense>
            </Route>
            <Route path="/player">
              <Suspense fallback={<Loading message="Loading Player" />}>
                <PlayerLazy />
              </Suspense>
            </Route>
          </Auth>
        </Switch>
      </Router>
    </UserContext.Provider>
  );
};

export default App;
