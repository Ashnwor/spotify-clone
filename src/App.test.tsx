import React from 'react';
import { shallow } from 'enzyme';
// eslint-disable-next-line import/extensions
import App from './App';

it('runs correctly', () => {
  const wrapper = shallow(<App />);

  expect(wrapper).toMatchSnapshot();
});
