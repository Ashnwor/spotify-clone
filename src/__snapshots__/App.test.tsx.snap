// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs correctly 1`] = `
<div>
  <Navbar />
  <Topbar />
  <ContextProvider
    value={
      Object {
        "currentSong": Object {},
        "setCurrentSong": [Function],
      }
    }
  >
    <ContextProvider
      value={
        Object {
          "setVolume": [Function],
          "volume": 1,
        }
      }
    >
      <Nowplaying />
    </ContextProvider>
    <Main />
  </ContextProvider>
</div>
`;
